scenes/World/World.gd

```diff
@@ -23,17 +22,27 @@ func _on_Dialog_timeline_end(_timeline_name) -> void:
 func _on_Dialog_dialogic_signal(value) -> void:
 	if value == "Yam":
 		var harrier = load("res://scenes/Ships/Harrier/Harrier.tscn")
-		GameState.player.ship = harrier
```

Comment:
Storing the ship classes in the game state was difficult to serialize to JSON and was redundant with the ship_name since there is a one to one mapping between the ship name and the godot class.

I changed this to instead store only the ship name (as a string) in the GameState and then created `_get_ship_class_from_name` function to get the class from the name when needed

Reply:
Aaron, notice the player_ship variable on line 4 of World.gd and it's use twice in the _on_Dialog_dialogic_signal function (line 27 and line 32). Do you think these lines of code are still necessary given the changes you've made to how ship information is stored and instanced in?

Reply:
Good catch, I pushed a new commit that removed that code

Reply:
Aaron, I think you may have forgotten to delete line 4. I don't think we need that anymore. It's approved but if you want change that so we don't have to remember to do it in another merge.